### PR TITLE
feat: migration typescript chart stepper statusindicator

### PR DESCRIPTION
**Pull request by vincentaudebert**
## Type

- Typescript Migration

Chart
Stepper
StatusIndicator

### CODE REVIEW

File: src/components/Chart/Donut.tsx [hunk 1]
```diff
@@ -81,9 +97,19 @@ const StyledContent = styled.div`
...
-const Donut = ({
-  height,
-  width,
+type DonutProps = {
+  chartId?: string
+  content?: React.ReactNode
+  data: Data[]
+  focused?: number
+  height?: number
+  onFocusChange(...args: unknown[]): unknown
```
Comment (chambo-e) [inline src/components/Chart/Donut.tsx:106]: FocusEventHandler maybe ?

File: src/components/Chart/Tooltip.tsx [hunk 2]
```diff
@@ -22,8 +23,21 @@ const Space = styled.span`
...
 `
 
-const Tooltip = ({ data }) => (
-  <div tabIndex="-1" role="tooltip">
+type TooltipProps = {
+  data: {
+    name?: string | null
+    value?: string | null
+    details?:
+      | {
```
Comment (chambo-e) [inline src/components/Chart/Tooltip.tsx:31]: Uh ?

File: src/components/Chart/__tests__/__snapshots__/index.tsx.snap [hunk 3]
```diff
@@ -48,11 +48,11 @@ exports[`Chart renders correctly when chart is hovered 1`] = `
   stroke-linecap: butt;
   fill: none;
   cursor: pointer;
-  stroke-dasharray: 566 566 10 0;
-  -webkit-transform: rotate(0deg);
-  -moz-transform: rotate(0deg);
-  -ms-transform: rotate(0deg);
-  transform: rotate(0deg);
+  stroke-dasharray: 566 10 0;
+  -webkit-transform: rotate(deg);
```
Comment (chambo-e) [inline src/components/Chart/__tests__/__snapshots__/index.tsx.snap:52]: :doubt:

File: src/components/Stepper/index.tsx [hunk 4]
```diff
@@ -104,62 +126,79 @@ const StyledContainer = styled(Box, {
...
-  name,
-  onBlur,
+type StepperProps = {
+  disabled?: boolean
+  maxValue?: number
+  minValue?: number
+  name?: string
+  onChange?(data: unknown): void
+  onMaxCrossed?(...args: unknown[]): unknown
+  onMinCrossed?(...args: unknown[]): unknown
```
Comment (chambo-e) [inline src/components/Stepper/index.tsx:145]:
```suggestion
  onMaxCrossed?(): void
  onMinCrossed?(): void
```

File: src/components/Stepper/index.tsx [hunk 5]
```diff
@@ -104,62 +126,79 @@ const StyledContainer = styled(Box, {
...
-  maxValue,
-  minValue,
-  name,
-  onBlur,
+type StepperProps = {
+  disabled?: boolean
+  maxValue?: number
+  minValue?: number
+  name?: string
+  onChange?(data: unknown): void
```
Comment (chambo-e) [inline src/components/Stepper/index.tsx:143]:
```suggestion
  onChange?(input: number): void
```

File: src/components/Stepper/index.tsx [hunk 6]
```diff
@@ -104,62 +126,79 @@ const StyledContainer = styled(Box, {
...
+  text = '',
   value,
   ...props
 }) => {
-  const inputRef = useRef()
-  const [inputValue, setInputValue] = useState(parseIntOr(value, minValue))
+  const inputRef =
+    useRef<HTMLInputElement>() as MutableRefObject<HTMLInputElement>
+  const [inputValue, setInputValue] = useState(
+    parseIntOr(value as string, minValue),
```
Comment (chambo-e) [inline src/components/Stepper/index.tsx:170]: No need to parse when value is already a number no ?

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: src/components/Stepper/index.tsx (hunk 6 above)
Comment (vincentaudebert) [reply src/components/Stepper/index.tsx:170]: Sadly value is `var value: string | number | readonly string[] | undefined`

File: src/components/Chart/Donut.tsx (hunk 1 above)
Comment (vincentaudebert) [reply src/components/Chart/Donut.tsx:106]: No it's a bit like the `onChange` on Stepper, it's a custom made one.

I updated though

File: src/components/Stepper/index.tsx (hunk 6 above)
Comment (chambo-e) [reply src/components/Stepper/index.tsx:170]: You could do 
```suggestion
    typeof value === "string" ? parseIntOr(value, minValue) : value,
```

**Review by chambo-e — APPROVED** [review]

**Pull request merged**